op-interop-mon/monitor/job.go

```diff
@@ -206,3 +211,29 @@ func (j *Job) DidMetrics() bool {
 func (j *Job) SetDidMetrics() {
 	j.didMetrics.Store(true)
 }
+
+// AddInitiatingHash adds a hash to the initiatingHash slice if it hasn't been seen before
+func (j *Job) AddInitiatingHash(hash common.Hash) {
+	j.rwLock.Lock()
+	defer j.rwLock.Unlock()
+
+	// Check if hash already exists
+	for _, existingHash := range j.initiatingHash {
+		if existingHash == hash {
+			return
+		}
+	}
```

Comment:
Could we just check against the most recent entry? If we went from hashA -> hashB -> hashA that might be useful to record and log out.

Reply:
True yeah. The metric collector will see it the same either way. I had wanted to avoid excess data if the hash flipped back and forth, but this is a good point, we might want to see it.